gsi/verification/jwt.py

```diff
@@ -0,0 +1,224 @@
+from collections.abc import Mapping
+import datetime
+import json
+
+import cachetools
+import six
+from six.moves import urllib
+
+from gsi.verification import _helpers
+from gsi.verification import crypt
+
+try:
+    from gsi.verification.crypt import es256
+except ImportError:  # pragma: NO COVER
+    es256 = None
+
+_ALGORITHM_TO_VERIFIER_CLASS = {"RS256": crypt.RSAVerifier}
+_CRYPTOGRAPHY_BASED_ALGORITHMS = frozenset(["ES256"])
+
+if es256 is not None:  # pragma: NO COVER
+    _ALGORITHM_TO_VERIFIER_CLASS["ES256"] = es256.ES256Verifier
+
+
+def encode(signer, payload, header=None, key_id=None):
```

Comment:
I kept the encode functionality to make jwt.py just a bit more general its abilities, but if this might be a source of confusion then I can move it to the test directory

Reply:
It should be ok since the developers will not be required to use the methods here directly.